src/cli/commands/install.js

```diff
@@ -284,7 +284,7 @@ export class Install {
       pushDeps('devDependencies', projectManifestJson, {hint: 'dev', optional: false}, !this.config.production);
       pushDeps('optionalDependencies', projectManifestJson, {hint: 'optional', optional: true}, true);
 
-      if (this.config.workspaceRootFolder) {
+      if (this.config.workspaceRootFolder && !this.ignoreWorkspaces) {
```

Comment:
Hm, doesn't this break running `yarn install` at the root of the workspace? It won't install the workspaces dependencies anymore, and its own dependencies won't have the same layout than what they would have been if the hoister had been aware of the workspaces.

Reply:
`this.ignoreWorkspaces` is only set on `add` (defaults to false), so install should be ok. I'd also imagine this test (https://github.com/yarnpkg/yarn/blob/master/__tests__/commands/install/workspaces-install.js#L71) would catch it?

Reply:
Oh I see! Ok, I missed that `ignoreWorkspaces` was undefined in `Install`. I think you should make it a flag defined in `Install` (rather than `Add`) and add a setter to toggle it (like the `topLevelBinLinkFlag` implemented [here](https://github.com/yarnpkg/yarn/pull/4265/files#diff-cebc41615330398e66d604a4a42bcd0e)).

Reply:
you got it!